tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/temporal/TableauEncoder.scala

```diff
@@ -0,0 +1,423 @@
+              .map(arg => {
+                encodeSyntaxTreeInPredicates(arg)
+              })
+              .unzip3
+
+            val argsPredsUnion = argsPreds.foldLeft(genericPredExs)(_ ++ _)
+
+            /* encode the node itself
+             */
+            oper match {
+              case TlaTempOper.box | TlaTempOper.diamond => /* curNode has the form []A or <>A */
+                /* create new auxiliary variable curNode_unroll or curNode_finally */
+                val nameSuffix = oper match {
+                  case TlaTempOper.box     => TableauEncoder.BOX_SUFFIX
+                  case TlaTempOper.diamond => TableauEncoder.DIAMOND_SUFFIX
+                }
+                val auxVarDecl = new TlaVarDecl(nodeIdentifier + nameSuffix)(Typed(BoolT1))
+
+                val auxVarEx = builder.varDeclAsNameEx(auxVarDecl)
+                val auxVarExPrime = builder.prime(auxVarEx)
+
+                /*
+                  initialize __temporal_curNode
+
+                  /\ __temporal_curNode \in BOOLEAN
+                  /\ __temporal_curNode_globally = TRUE or /\ __temporal_curNode_finally = FALSE
+                 */
+                val initialValue = oper match {
+                  case TlaTempOper.box     => true
+                  case TlaTempOper.diamond => false
+                }
+                val auxVarInitEx = builder.eql(auxVarEx, builder.bool(initialValue))
+
+                /* update __temporal_curNode:
+
+                  /\ __temporal_curNode' \in BOOLEAN
+                  /\ __temporal_curNode_globally' \in BOOLEAN or __temporal_curNode_finally' \in BOOLEAN
+
+                                           outerOp                                          outerOp
+                  /\ __temporal_curNode <=> A /\ __temporal_curNode' or __temporal_curNode <=> A \/ __temporal_curNode'
+
+                                                    outerOp
+                  /\ __temporal_curNode_globally' <=> /\ __temporal_curNode_globally
+                                                      /\ (~InLoop' \/ A')
+                                                    outerOp      innerOp
+                  or
+                                                   outerOp
+                  /\ __temporal_curNode_finally' <=>  \/ __temporal_curNode_finally
+                                                      \/  (InLoop' /\ A')
+                                                    outerOp      innerOp
+                 */
+
+                /*
+                box and diamond differ in the inner and outer operators,
+                and whether the InLoop variable is negated in the conjunction
+                 */
+                val (outerOpTmp, innerOpTmp, innerInLoopEx) = oper match {
+                  case TlaTempOper.box     => (builder.and _, builder.or _, builder.not(loopEnc.inLoopPrime))
+                  case TlaTempOper.diamond => (builder.or _, builder.and _, loopEnc.inLoopPrime)
```

Comment:
> This gives me the following error: "missing argument list for method and in trait BoolBuilder Unapplied methods are only converted to functions when a function type is expected. You can make this conversion explicit by writing `and _` or `and(_)` instead of `and`."

You can fix this by annotating the LHS IIRC.